API/src/main/java/de/brandwatch/minianalytics/api/controller/MentionController.java

```diff
@@ -2,64 +2,63 @@
+import org.springframework.web.server.ResponseStatusException;
 
-import java.time.Instant;
+import java.time.DateTimeException;
+import java.time.LocalDate;
 import java.time.LocalDateTime;
-import java.time.ZoneId;
-import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
-import java.time.temporal.TemporalAccessor;
-import java.util.List;
+import java.time.ZoneOffset;
 
 @RestController
 public class MentionController {
 
-    private static final Logger logger = LoggerFactory.getLogger(RestController.class);
+    private static final Logger logger = LoggerFactory.getLogger(MentionController.class);
 
     private static final Gson gson = new Gson();
 
-    private final MentionRepository mentionRepository;
+    private final MentionService mentionService;
 
     @Autowired
-    public MentionController(MentionRepository mentionRepository) {
-        this.mentionRepository = mentionRepository;
+    public MentionController(MentionService mentionService) {
+        this.mentionService = mentionService;
+        LocalDate.now().minusDays(1).toString();
     }
 
     @GetMapping(value = "/mentions/{queryID}")
     public ResponseEntity findMentionsFromQuery(
             @PathVariable String queryID,
-            @RequestParam(value = "date", defaultValue = "") String date){
-
-        logger.info("GET /mentions/" + queryID);
-
-        List<Mention> mentions = mentionRepository.findByQueryID(Long.parseLong(queryID));
-
-        if(!date.equals("")){
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-
-            String convertedDate = date + " 00:00:00";
-            TemporalAccessor temporalAccessor = formatter.parse(convertedDate);
-            LocalDateTime localDateTime = LocalDateTime.from(temporalAccessor);
-            ZonedDateTime zonedDateTime = ZonedDateTime.of(localDateTime, ZoneId.systemDefault());
-            Instant instant = Instant.from(zonedDateTime);
-
-            String dateBounds = "[" + instant + " TO *]";
-
-            logger.info("queryID: *?0* AND date: *?1*");
-
-            return ResponseEntity.status(200).body(gson.toJson(mentionRepository.findMentionsAfterDate(
-                    Long.parseLong(queryID), dateBounds)));
+            @RequestParam(value = "startDate", defaultValue = "")
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime startDate,
+            @RequestParam(value = "endDate", defaultValue = "")
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime endDate) {
+        try{
+            if(startDate == null) startDate = LocalDateTime.now().minusDays(1);
+            if(endDate == null) endDate = LocalDateTime.now();
+
+            if(!validateDates(startDate, endDate)){
+                throw new DateTimeException("Dates are not in the right order");
+            }
+
+            return ResponseEntity.status(200).body(gson.toJson(mentionService.getMentionsFromQueryID(queryID,
```

Comment:
Do you know for a `RestController` you don't have to create ResponseEntity and transform object to json yourself unless you want some non-standard behaviour, it can be done automatically, e.g. https://www.baeldung.com/spring-controller-vs-restcontroller
Would be a little less boilerplate

Reply:
Didnt know. But I changed it.